src/makeInfinityAdder.js

```diff
@@ -6,7 +6,22 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(x) {
+    if (x !== undefined) {
+      sum += x;
+      return adder;
+    }
+
+    if (x === undefined) {
+      const result = sum;
+      sum = 0;
+      return result;
+    }
```

Comment:
Since you are checking  `x !== undefined` above (and returning inside that if statement), you don't another if here. 
```suggestion
      const result = sum;
      sum = 0;
      return result;
```